bin/dothething

```diff
@@ -30,147 +34,247 @@ rescue OptionParser::InvalidOption => e
  exit 1
 end.parse!
 
-class State
- #Loads the config file
- def initialize(state)
-  @statefile = state 
-  @sf = YAML.load_file(@statefile)
- end
+#If no command was entered display help and exit 
+if options.empty?
+ puts opt_parser
+ exit 1
+end
 
- #Toggles value in state hash
- def toggle(name)
-  #Finds all true/false values in hash
-  t = @sf[name].key(true)
-  f = @sf[name].key(false) 
+module ConfigFile
+ class Config
+  #Load the config file
+  def initialize
+   configfile = File.join($path, 'config.yml')
+   @config = YAML.load_file(configfile)
+  end
+  
+  #Check if a setting has been set
+  def setting_exists?(setting)
+   if @config[setting] != nil then
+    return true 
+   else
+    puts "The config file does not contain a path for the setting: #{setting}"
+    exit 1 
+   end 
+  end
+  
+  #Set the directory for the job   
+  def jobdir(name)
+   setting_exists?('jobdir')
+   @jobdir = @config['jobdir']
+   
+   #Check if the path given in the file is absolute or relative   
+   if @jobdir.include? "../"
+    @jobdir = File.join($path, @jobdir, name)
+   else
+    @jobdir = File.join(@jobdir, name)
+   end      
+  end
+  
+  #Check if a file exists
+  def file_exist?(setting)
+   if File.exist?(@config[setting])
+    return true
+   else
+    #Create the file if it doesn't
+    system "touch", "#{@config[setting]}"
+    return false
+   end
+  end
   
-  #Toggles the values
-  @sf[name][:"#{t}"] = false
-  @sf[name][:"#{f}"] = true  
+  #Check if the log file setting and file exist
+  #If they do then it returns the set path
+  def logfile
+   setting_exists?('logfile')
+   file_exist?('logfile')
+   @config['logfile']
+  end
+  
+  #Check if the state file setting and file exist
+  #If they do then it returns the set path
+  def statefile
+   setting_exists?('statefile')
+   if file_exist?('statefile') == false
+    #Create the necessary configuration within state.yaml
+    #This only runs if the file was just created within file_exist?
+    open(@config['statefile'], 'a') { |f|
```

Comment:
This would be better done by just calling to_yaml on a hash and dumping that to the file, eg
`File::write_file(@config.statefile,[:job1=>[:forward=>true,:rewind=>false]].to_yaml)`
Saying that though, its not really needed at all as an empty file should be supported by your state class.

Reply:
The state class should now be able to handle empty files, creating the necessary structure if the job has been validated as well as appending new jobs (e.g. job2) to the yaml file